codegen-test/model/rest-json-extras.smithy

```diff
@@ -78,7 +78,8 @@ service RestJsonExtras {
         code: 500,
         body: "",
         headers: { "X-Amzn-Errortype": "ExtraError" },
-        params: {}
+        params: {},
+        appliesTo: "client",
```

Comment:
This test fails for the server because [servers are expected to serialize `{}` when the output parameters are empty](https://github.com/awslabs/smithy/blob/69cc7ba87b62da026130876d3e306194f39759ee/smithy-aws-protocol-tests/model/restJson1/errors.smithy#L32-L36).

Hence why we have a [`serverStructureSerializer`](https://github.com/awslabs/smithy-rs/blob/6c21fb0eb377c7120a8179f4537ba99a4b50ba96/codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/serialize/JsonSerializerGenerator.kt#L173-L173) that behaves like that. We should arguably modify the Kotlin docs for `serverStructureSerializer` calling this out.

Anyway, this test can remain for the client only, but we should duplicate in a new test that only applies to the server and that has `body: "{}"`, and copy over the documentation from `RestJsonGreetingWithErrors` to justify why we're doing so.